speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp

```diff
@@ -1,4 +1,134 @@
+#include <ostream>
 #include "FileDataStorage.h"
-#include "FileDataStorage.inl"
 
-CREATE_FILE_DATA_STORAGE_IMPLEMENTATION(BlackBoxData)
+// Write a literal string to the stream, as the string in text format (without conversion)
+#define WRITE_STRING_LIT(stream, string) stream << string << "\n"
+// Write a variable string to the stream, as the string in text format (without conversion)
+#define WRITE_STRING_VAR(stream, string) stream << string << "\n"
+// Write a variable to the stream
+#define WRITE_VAR(stream, val) stream << std::to_string(val) << "\n" //Binary: stream.write(reinterpret_cast<const char*>(&val), sizeof(val)); stream << "\n"
+
+/// @brief Write a time variable to a stream as a DateTime entry (aka as a "YYYY-MM-DD hh:mm:ss" string),
+///  finished by a newline.
+/// @param stream Output stream to write time to.
+/// @param date Time to format and write to the stream.
+inline void writeTime(std::ostream& stream, time_t date) {
+    // "YYYY-MM-DD hh:mm:ss" is 19 characters, finishing with a nullpointer makes 20.
+    // Thus allocate space for 20 characters.
+    char buffer[20];
+    strftime(buffer, 20, "%F %T", gmtime(&date));
+    stream << buffer << "\n";
+}
+
+/// @brief Initialize the file data storage.
+/// End result: a file is created at the given filepath, and initial data is written to the file.
+/// @param p_saveSettings Settings for what data to store.
+/// @param p_fileName Path of the file to save.
+/// @param p_userId User ID of the current player.
+/// @param p_trialStartTime Start time of the current race
+/// @param p_blackboxFilename Filename without path (e.g. "blackbox.exe") for the current black box
+/// @param p_blackboxName Name of the current black box (e.g. "Client Black Box")
```

Comment:
```suggestion
/// @param p_blackboxName Name of the current black box (e.g. "Lane Keeping Black Box")
```
Client blackbox can be confusing for people new to the project. (If there is a client black box what is the server black box?)